### PR TITLE
Feature/#7 test trained svm

### PR DESCRIPTION
**Pull request by palucki**
Added json file with test data (slightly modified training data)
Moved some common functionality to` common.h`
Added `test_svm_with_data(const CvSVM& svm, const nlohmann::json& test_data_json)`
and a simple test program `PC/svm/svm_test/svm_test.cpp`

### CODE REVIEW

File: PC/svm/includes/common.h [hunk 1]
```diff
@@ -0,0 +1,73 @@
+#ifndef COMMON
+#define COMMON
```
Comment (Qbicz) [inline PC/svm/includes/common.h]: I would keep COMMON_H convention for header guards.

File: PC/svm/includes/common.h [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+{
+    std::vector<float> training_element;
+    if(debug)
+        std::cout << "element: " << element << std::endl; 
+    training_element.push_back(element["x"]);
+    training_element.push_back(element["y"]);
+    return training_element;
+}
+
+void print_vector(std::vector<float> &vect)
```
Comment (Qbicz) [inline PC/svm/includes/common.h:20]: We will have conflicts, as I already moved it to modules/print_vector.cpp on my branch. But alright, you were first ;)

File: PC/svm/includes/common.h [hunk 3]
```diff
@@ -0,0 +1,73 @@
+#ifndef COMMON
+#define COMMON
+
+#include "json.hpp"
+
+#define DATA_DIMENSIONS 2 // for Hu moments, it will be 7
+
+using json =nlohmann::json;
+
+std::vector<float> svm_build_single_element(json &element, bool debug = false)
```
Comment (Qbicz) [inline PC/svm/includes/common.h:10]: Is having implementations in .h file a good practice?

**Review by Qbicz — COMMENTED** [review]
Good job, we now have a solution for testing.

File: PC/svm/includes/common.h (hunk 1 above)
Comment (palucki) [reply PC/svm/includes/common.h]: Done

File: PC/svm/includes/common.h (hunk 3 above)
Comment (palucki) [reply PC/svm/includes/common.h:10]: Compilation times are longer, code is inlined (so binary is bigger). It should be moved to some .cpp but for now I will leave it.

**Pull request merged**